cartographer_ros/cartographer_ros/map_builder_bridge.h

```diff
@@ -59,7 +59,7 @@ class MapBuilderBridge {
       const std::shared_ptr<::cartographer_ros_msgs::srv::SubmapQuery::Request> request,
       std::shared_ptr<::cartographer_ros_msgs::srv::SubmapQuery::Response> response);
 
-  cartographer_ros_msgs::msg::SubmapList GetSubmapList();
+  cartographer_ros_msgs::msg::SubmapList GetSubmapList(rclcpp::node::Node::SharedPtr& node);
```

Comment:
If the timesource and clock are members you won't need to pass through the node.

Reply:
I ended up doing something slightly different here.  I made it so that we only initialize and store the `TimeSource` and the `Clock` once, in the top-level `Node` object.  Then I pass down a shared_ptr to the `Clock` for this call; that way, the entirety of cartographer_ros is running off of the same `TimeSource` and `Clock`.  Take a look and let me know what you think about this.

Reply:
That makes sense to pass through the smaller object pointer, that's already initialized. For that method there's a level that I wonder if it should be pulling the data from the internal structure instead of "now". Maybe querying for the latest timestamp in the stored data etc. For example if the input data had stopped coming in the map is from a state in the past not "now". But that's a more general question than this compile fix.

Reply:
That's an interesting point, now that you bring it up.  Just to make sure this wasn't an error I introduced when doing the initial port, I took a look at the ROS1 version of the code here: https://github.com/googlecartographer/cartographer_ros/blob/master/cartographer_ros/cartographer_ros/map_builder_bridge.cc#L170.  It is using `::ros::Time::now`, so if there is a problem, it is also an upstream one.  I'm going to merge this fix, and we can bring this problem up with upstream if we want to follow up.  Thanks for the review.